### PR TITLE
⭐ mql discovery for ecr images

### PR DESCRIPTION
**Pull request by vjeffrey**
introduces mql resource for images

as a followup, we can replace a lot of the code in the ecr.go discovery with the data we get back from mql
```
cnquery> aws.ecr.images { * }
aws.ecr.images: [
  0: {
    registryId: "172746783610"
    tags: [
      0: "latest"
    ]
    digest: "sha256:0c78b32ef7f3b41e3ed3115488d64a6faf7a3cdade2a5eb720092b6e8e0a88ca"
    repoName: "vjtestpriv"
    mediaType: "application/vnd.docker.distribution.manifest.v2+json"
  }
]
cnquery> aws.ecr.publicRepositories { * }
aws.ecr.publicRepositories: []
cnquery> aws.ecr.privateRepositories { * }
aws.ecr.privateRepositories: [
  0: {
    uri: "172746783610.dkr.ecr.us-east-1.amazonaws.com/vjtestpriv"
    public: false
    region: "us-east-1"
    registryId: "172746783610"
    name: "vjtestpriv"
    arn: "arn:aws:ecr:us-east-1:172746783610:repository/vjtestpriv"
    images: [
      0: aws.ecr.image id = vjtestpriv/sha256:0c78b32ef7f3b41e3ed3115488d64a6faf7a3cdade2a5eb720092b6e8e0a88ca
    ]
  }
]
cnquery> exit
```

### CODE REVIEW

File: resources/packs/aws/aws_ecr.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+			}
+			mqlres = append(mqlres, mqlImage)
+		}
+	} else {
+		svc := at.Ecr(region)
+		res, err := svc.DescribeImages(ctx, &ecr.DescribeImagesInput{RepositoryName: &name})
+		if err != nil {
+			return nil, err
+		}
+		for i := range res.ImageDetails {
```
Comment (czunker) [inline resources/packs/aws/aws_ecr.go:183]: I do not see any difference in the code, once you got the response from `DescribeImages`.
You could move this out of the `if public` to have the `for` loop only once.

**Review by czunker — COMMENTED** [review]
I wanted to try it locally, but it didn't work:
```
cnquery> aws.ecr{*}
failed to compile: cannot find field 'ecr' in aws
```

File: resources/packs/aws/aws_ecr.go (hunk 1 above)
Comment (vjeffrey) [reply resources/packs/aws/aws_ecr.go:183]: the only issue there is the types are different

**Comment by vjeffrey** [comment]
that's odd. i'll pull it down again and give it another try

File: resources/packs/aws/aws_ecr.go (hunk 1 above)
Comment (czunker) [reply resources/packs/aws/aws_ecr.go:183]: Oh, didn't see this. :+1:

**Comment by czunker** [comment]
> aws.ecr { * }

Ok, no clue what I did wrong before. Re-tested it and works as expected.

Thanks for giving it another try.

**Review by czunker — APPROVED** [review]
Thanks @vjeffrey

**Pull request merged**